test/Benchmarks/src/Table/Performance_Research/Column_Bi_Map.enso

```diff
@@ -0,0 +1,90 @@
+## NOTE
+   This benchmark assumes that the Table operations used here are implemented in Java.
+   This is a valid assumption as of 2023-07-12.
+   If we ever change these operations, we would need to add this benchmark to have custom implementations for the Java side as well if we still want to use it for comparisons.
+
+   This file is _not_ automatically run when the benchmarks are run.
+from Standard.Base import all
+from Standard.Table import all
+
+from Standard.Test import Bench
+
+import project.Table.Performance_Research.Common_Setup.Common_Setup
+import project.Table.Performance_Research.Helpers
+
+polyglot java import org.enso.table.data.column.builder.NumericBuilder
+polyglot java import org.enso.table.data.column.builder.StringBuilder
+
+# Adding two String columns
+type Boxed_Bi_Map_Test
+    Instance text_column_1 text_column_2
+
+    java_map self =
+        self.text_column_1 + self.text_column_2
+
+    enso_map_as_vector self convert_polyglot_dates =
+        vector_proxy_1 = self.text_column_1.to_vector
+        vector_proxy_2 = self.text_column_2.to_vector
+        mapped = vector_proxy_1.zip vector_proxy_2 (+)
+        Helpers.column_from_vector "result" mapped convert_polyglot_dates=convert_polyglot_dates
+
+    enso_map_with_builder self =
+        n = self.text_column_1.length
+        if self.text_column_2.length != n then Panic.throw "LENGTH MISMATCH" else
+            builder = StringBuilder.new n
+            storage_1 = self.text_column_1.java_column.getStorage
+            storage_2 = self.text_column_2.java_column.getStorage
+            0.up_to n . each i->
+                item_1 = storage_1.getItemBoxed i
+                item_2 = storage_2.getItemBoxed i
+                if item_1.is_nothing || item_2.is_nothing then builder.appendNulls 1 else
+                    res = item_1 + item_2
+                    builder.append res
+            Column.from_storage "result" builder.seal
+
+# Adding two Long columns
+type Primitive_Bi_Map_Test
+    Instance int_column_1 int_column_2
+
+    java_map self =
+        self.int_column_1 + self.int_column_2
```

Comment:
I cannot imagine what it means to `+` one column with another. Maybe I could find that out in a debugger, but certainly not from a review in a browser.

Reply:
As in the docs, `+` adds values in the two columns, row-by-row.

Or if you mean _how_ it is implemented - you can trace it down in `Column.enso` and then `Storage.java`, `AbstractLongStorage.java`. This is the 'Java vectorized' implementation, i.e. the implementation that iterates over a `long[]` array of the two storages and adds each rows together, creating a new array-backed storage.

Reply:
Please let me know if any further explanations are needed and what is unclear and I will do my best to explain it. We can also schedule a call if you prefer to talk.

Reply:
Thank you Radek. Yes, I should be looking at `Column.enso` and `Storage.java`.